### PR TITLE
{AKS} Refactor 1st part of the update sub-command in acs module

### PR DESCRIPTION
**Pull request by FumingZhang**
**Description**

* refactor basic structure of `aks_update`.
* refactor auto scaler profile.

Unit Test Results:

- "covered_lines": 1248,
- "num_statements": 1268,
- "percent_covered": 98.42271293375394,
- "missing_lines": 20,
- "excluded_lines": 0

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 1]
```diff
@@ -274,7 +328,7 @@ class AKSCreateContext:
     read_only is set to True when necessary to avoid loop calls, when using the getter function to obtain the value of
     other parameters.
     """
-    def __init__(self, cmd: AzCliCommand, raw_parameters: Dict):
+    def __init__(self, cmd: AzCliCommand, raw_parameters: Dict, mode=DecoratorMode.CREATE):
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:331]: the mode is hard to understand. maybe give it a full name like decoratorMode?

**Review by andyliuliming — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 2]
```diff
@@ -3929,3 +4077,166 @@ def create_mc(self, mc: ManagedCluster) -> ManagedCluster:
...
+        This function will record an indicator indicating the status of the update. The indicator is a boolean
+        intermediate value, once updated, it will be set to True. The purpose is to identify whether the `mc` object
+        has been updated.
+
+        :return: None
+        """
+        previous_state = self.context.get_intermediate(intermediate_name, default_value=False)
+        self.context.set_intermediate(intermediate_name, is_updated or previous_state, overwrite_exists=True)
+
+    def __check_is_updated(self, intermediate_name="updated") -> None:
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:4116]: could you please clarify more about this function?

**Review by andyliuliming — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py (hunk 1 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/decorator.py]: Good catch, changed in commit [#1897ad4](https://github.com/Azure/azure-cli/pull/19647/commits/1897ad4edc8e7262926591df1f5fdaaafffe8595).

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py (hunk 2 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/decorator.py]: We have changed the mechanism to check whether the parameters have been set in commit [#123aca0](https://github.com/Azure/azure-cli/pull/19647/commits/123aca085cd4d897e8b2949e53e1c6bdef4f3ad4).

**Review by andyliuliming — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**